jpegenc/subblocks/block_buffer/block_buffer.py

```diff
@@ -0,0 +1,181 @@
+#!/usr/bin/env python
+# coding=utf-8
+
+from myhdl import Signal, always_comb, always_seq, block, intbv, ResetSignal
+from jpegenc.subblocks.common import ram_in, ram_out, block_buffer_in, block_buffer_out
+from myhdl.conversion import analyze
+from math import log2, ceil
+from jpegenc.subblocks.block_buffer import RAM
+from jpegenc.subblocks.common import assign
+
+@block
+def block_buffer(inputs, outputs, clock, reset, line_width=16):
+
```

Comment:
Definitely it need a lot of documentation this block. It is in experimental stage yet. The line_width is used to define the size of the 8 rams.